# Bioinformatics-Genomics-learning-resource

### Books
Bioinformatics Data Skills: Reproducible and Robust Research with Open Source Tools https://vincebuffalo.com/book/. The scripts for the book can be found [here](https://github.com/vsbuffalo/bds-files/)

[Python for Biologists](https://pythonforbiologists.com/)

### Reads 

[Cancer Genomics, Gene Expression data analysis using Python](https://medium.com/adventures-in-healthcare-data/cancer-genomics-ii-exploring-biomarkers-of-liver-cancer-in-gene-expression-data-using-python-b719b519b406)

### Courses 

[Command Line Tools for Genomic Data Science- Coursera](https://www.coursera.org/learn/genomic-tools)

### Linux Command in NGS Data Analysis 

Check the Linux_command_in_NGS.md file for more examples

### Resources for common tools in NGS

##### FASTQC 

https://rtsf.natsci.msu.edu/genomics/tech-notes/fastqc-tutorial-and-faq/
[FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/), written by Simon Andrews of Babraham Bioinformatics, is a very popular tool used to provide an overview of basic quality control metrics for raw next generation sequencing data. 


There are a number different analyses (called modules) that may be performed on a sequence data set. 

The authors give good examples of why certain metrics may fail

[FATSQC Help](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/)




